app/src/componentLibrary/checkbox/checkbox.scss

```diff
@@ -0,0 +1,130 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@mixin checkboxContainer($color, $opacity) {
+  display: inline-flex;
+  align-items: center;
+  padding-left: 24px;
+  color: $color;
+  font-family: $FONT-ROBOTO-REGULAR;
+  font-size: 13px;
+  line-height: 20px;
+  .disabled {
+    @include disabled-state($opacity);
+  }
+}
+
+@mixin checked-state($borderColor, $backgroundColor) {
+  border-color: $borderColor;
+  background-color: $backgroundColor;
+  background-repeat: no-repeat;
+  background-position: center;
+  background-image: url("data:image/svg+xml,%3Csvg width='12' height='9' viewBox='0 0 12 9' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M1.5 4.25L4.5 7.25L10.5 0.75' stroke='white' stroke-width='2'/%3E%3C/svg%3E");
+}
+
+@mixin disabled-state($opacity) {
+  opacity: $opacity;
+  pointer-events: none;
+}
+
+.light-container {
+  @include checkboxContainer($COLOR--almost-black, 0.3);
+}
+
+.input {
+  display: none;
+}
+
+.checkbox {
+  position: absolute;
+  width: 16px;
+  height: 16px;
+  border: 1px solid $COLOR--e-200;
+  border-radius: 3px;
+  margin-left: -24px;
+  box-sizing: border-box;
+  cursor: pointer;
+}
+
+.children-container {
+  cursor: pointer;
+}
+
+.input:checked + .light {
+  @include checked-state($COLOR--topaz-2, $COLOR--topaz-2);
+}
+
+.input:hover + .light {
+  border-color: $COLOR--e-400;
+}
+
+.input:active:not(:disabled) + .light {
+  border: 2px solid $COLOR--topaz-focused;
+}
+
+.input:checked:hover + .light {
+  @include checked-state($COLOR--topaz-hover-2, $COLOR--topaz-hover-2);
+}
+
+.input:checked:active:not(:disabled) + .light {
+  @include checked-state($COLOR--topaz-focused, $COLOR--topaz-focused);
+}
+
+.input:disabled + .light {
+  border-color: $COLOR--e-200;
+}
+
+.input:checked:disabled + .light {
+  @include checked-state($COLOR--topaz-2, $COLOR--topaz-2);
+  @include disabled-state(0.3);
+}
+
+.dark-container {
+  @include checkboxContainer($COLOR--darkmode-gray-100, 0.5);
+}
+
+.dark {
+  border-color: $COLOR--darkmode-gray-100;
+}
+
+.input:checked + .dark {
+  @include checked-state($COLOR--darkmode-topaz-main, $COLOR--darkmode-topaz-main);
+}
+
+.input:hover + .dark {
+  border-color: $COLOR--darkmode-gray-50;
+}
+
+.input:active:not(:disabled) + .dark {
+  border: 2px solid $COLOR--darkmode-topaz-focused;
+}
+
+.input:checked:hover + .dark {
+  @include checked-state($COLOR--darkmode-topaz-hover, $COLOR--darkmode-topaz-hover);
+}
+
+.input:checked:active:not(:disabled) + .dark {
+  @include checked-state($COLOR--darkmode-topaz-focused, $COLOR--darkmode-topaz-focused);
+}
+
+.input:disabled + .dark {
+  border-color: $COLOR--darkmode-gray-100;
+}
+
+.input:checked:disabled + .dark {
```

Comment:
It would be better to organize the variant styles by one common parent selector, f.e.
```css
.dark {
   .input {
      &:checked {}
      
      &:disabled {}
      
      ...
   }
}
```

Also inspired by the dropdown component.
And will be easier to edit and add new variants with such implementation.

Reply:
@oranje322 I'll merge it, could you please update this in another PR?